build: poetry migration (#98)
canonical/pvcviewer-operator: CONTRIBUTING.md

### Updating Dependencies

To add/update/remove any dependencies and/or to upgrade Python, simply:

1. first add/update/remove such dependencies to/in/from the desired group(s) below `[tool.poetry.group.<your-group>.dependencies]` in `pyproject.toml`, and/or upgrade Python itself in `requires-python` under `[project]`

    _⚠️ dependencies for the charm itself are also defined as dependencies of a dedicated group called `charm`, specifically below `[tool.poetry.group.charm.dependencies]`, and not as project dependencies below `[project.dependencies]` or `[tool.poetry.dependencies]` ⚠️_

2. then run `tox -e update-requirements` to update not only the lock file but also the exported `requirements.txt`-like charm dependencies used by `charmcraft`

3. optionally, if you also want to update your local environment for running Python commands/scripts yourself and not through tox, see [Running Python Environments](#running-python-environments) below

> By point 2., `poerty`, through `tox`, will let you know if there are any dependency conflicts to solve.

Reviewer: nitpick: Potentially include this information in point 2. itself. Feel free to ignore:
```suggestion
```

### Running `tox` Environments

To run `tox` environments locally, ensure to have `tox` installed first and then simply run your `tox` environments natively (e.g.: `tox -e lint`). `tox` will internally first install `poetry` and then rely on it to install and run its environments.

### Running Python Environments

> To run Python commands/scripts locally from any environments built from any combinations of dependency groups without relying on `tox`:
> 1. make sure you have `poetry` installed
> 1. install any dependency groups that compose the environment of interest: `poetry install --only <your-group-a>,<your-group-b>` (or all groups, if you prefer: `poetry install --all-groups`)
> 2. run Python commands/scripts via poetry: `poetry run python3 <whatever>`

Reviewer: nitpick: Try to make less conversational, also fix the order of bullet points (there are two "1"s currently)
```suggestion
To run Python scripts locally from any environments built from any combinations of dependency groups without relying on `tox`:
1. Ensure you have `poetry` installed
1. Install all requirements for the dependency groups you need: `poetry install --only <your-group-a>,<your-group-b>` (or all groups, if you prefer: `poetry install --all-groups`)
2. Run Python scripts via poetry: `poetry run python3 <script>`
```
